lib/src/charts/painters/cartesian/cartesian_chart_painter.dart

```diff
@@ -57,10 +61,10 @@ class CartesianChartPainter extends CustomPainter {
     // Finally for every data series, we will construct a painter and handover
     // the canvas to them to draw the data sets into the required chart
     for (var series in controller.currentData) {
-      // construct the painter for this data
-      var painter = paintBuilder(series.runtimeType);
+      // get the painter for this data
+      var painter = controller.painters[series.runtimeType];
       // and paint the chart for given series
-      painter.paint(series, canvas, this);
+      painter?.paint(series, canvas, this);
```

Comment:
If painter is null, it could be difficult to debug what is wrong.
Either throw an error similar to what we do when we get config or atleast log an error to console that the painter was not found.
Ideally this painter should never be null, so I would say throwing here would be preferable

Reply:
add null check and threw Argument error if fails